File: src/main/java/com/tom/student/Student.java
package com.tom.student;

public class Student {
    String id;
    String name;
    int english;
    int math;

    public Student(String name,int english,int math) {
        this.name = name;
        this.english = english;
        this.math = math;
    }

    public int highest(){
//        int max=0;
        /*if (english>math) {
            max=english;
        }else {
            max=math;
        }*/
        return (english >math) ? english:math;
    }
    public void print() {
        int average = getAverage();
        System.out.print(name +"\t"+ english+"\t"+ math +
                "\t" + getAverage() + "\t" +
                ((getAverage() >= 60) ? "Pass" : "Failed"));
        char grading = 'F';
        switch (average/10){
            case 10:
            case 9:
                grading ='A';
                break;
            case 8:
                grading ='B';
                break;
            case 7:
                grading ='C';
                break;
            case 6:
                grading ='D';
                break;
                default:
                grading='F';
        }
        System.out.println("\t"+grading);
        /*if (getAverage() >= 60){
            System.out.println("\tPASS");
        }else {
            System.out.println("\tFailed");
        }*/
    }
    public int getAverage(){
        return (english+math)/2;
    }
}
 

File: src/main/kotlin/com/tom/student/student.kt
package com.kotlin

import java.util.*

fun main() {
//    userinput()
    val stu = Student("Jack", 70, 80)
    stu.print()
//    println("High score:"+ stu.higest())
    println("High score:${stu.higest()}") /* $?????? */
}

private fun userinput() {
    val scanner = Scanner(System.`in`)
    print("Please Enter Studnet name:")
//    var name=scanner.next()
    var name = null
    print("Please Enter Studnet english:")
    var english = scanner.nextInt()
    print("Please Enter Studnet math:")
    var math = scanner.nextInt()
    val stu = Student(name, english, math)
    stu.print()
    stu.nameChack()
}

class Student(var name:String?,var english:Int,var math:Int){
    fun print(){
        println("$name\t$english\t$math\t${getAverge()}\t${passOrFailed()}\t${grading()}")
        /*print(name+"\t"+english+"\t"+math+"\t"
                +getAverge()+"\t"+
                if(getAverge()>=60) "Pass" else "Failed")
        println("\t"+grading())*/
    }
    fun grading():Char{
        return when(getAverge()){
            in 90..100 ->'A'
            in 80..89 ->'B'
            in 70..79 ->'C'
            in 60..69 ->'D'
            else ->'F'
        }
        /*var grading:Char =when(getAverge()){
            in 90..100 ->'A'
            in 80..89 ->'B'
            in 70..79 ->'C'
            in 60..69 ->'D'
            else ->'F'
        }
        return grading;*/
    }
    fun passOrFailed()=
            if(getAverge()>=60) "Pass" else "Failed"
    fun getAverge()=(english+math)/2

    fun higest():Int {
        var max=
        return if (english>math){
            println("english")
            english
        }else{
            println("math")
            math
        }
    }

    fun nameChack(){
        println(name?.length)
    }

}